streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -521,7 +524,7 @@ protected Map<String, Integer> getNumPartitions(final Set<String> topics,
         for (final Map.Entry<String, KafkaFuture<TopicDescription>> topicFuture : futures.entrySet()) {
             final String topicName = topicFuture.getKey();
             try {
-                final TopicDescription topicDescription = topicFuture.getValue().get();
+                final TopicDescription topicDescription = topicFuture.getValue().get(Long.parseLong(DEFAULT_API_TIMEOUT_MS_CONFIG), TimeUnit.MILLISECONDS);
```

Comment:
Do not put timeout parameters on Future.get. This will not abort the operation. If you want timeouts, they should be done by adjusting your admin client configuration parameters.

Reply:
removed